torch/csrc/api/include/torch/nn/functional/upsampling.h

```diff
@@ -42,65 +48,75 @@ inline Tensor interpolate(const Tensor& input, InterpolateOptions options) {
+  } else if (input.dim() == 3 && c10::get_if<enumtype::kLinear>(&mode)) {
+    return torch::upsample_linear1d(input, _output_size(1), *align_corners);
+  } else if (input.dim() == 3 && c10::get_if<enumtype::kBilinear>(&mode)) {
     TORCH_CHECK(false, "Got 3D input, but bilinear mode needs 4D input");
-  } else if (input.dim() == 3 && c10::get_if<enumtype::kTrilinear>(&options.mode())) {
+  } else if (input.dim() == 3 && c10::get_if<enumtype::kTrilinear>(&mode)) {
     TORCH_CHECK(false, "Got 3D input, but trilinear mode needs 5D input");
-  } else if (input.dim() == 4 && c10::get_if<enumtype::kLinear>(&options.mode())) {
+  } else if (input.dim() == 4 && c10::get_if<enumtype::kLinear>(&mode)) {
     TORCH_CHECK(false, "Got 4D input, but linear mode needs 3D input");
-  } else if (input.dim() == 4 && c10::get_if<enumtype::kBilinear>(&options.mode())) {
-    return torch::upsample_bilinear2d(input, _output_size(2), *options.align_corners());
-  } else if (input.dim() == 4 && c10::get_if<enumtype::kTrilinear>(&options.mode())) {
+  } else if (input.dim() == 4 && c10::get_if<enumtype::kBilinear>(&mode)) {
+    return torch::upsample_bilinear2d(input, _output_size(2), *align_corners);
+  } else if (input.dim() == 4 && c10::get_if<enumtype::kTrilinear>(&mode)) {
     TORCH_CHECK(false, "Got 4D input, but trilinear mode needs 5D input");
-  } else if (input.dim() == 5 && c10::get_if<enumtype::kLinear>(&options.mode())) {
+  } else if (input.dim() == 5 && c10::get_if<enumtype::kLinear>(&mode)) {
     TORCH_CHECK(false, "Got 5D input, but linear mode needs 3D input");
-  } else if (input.dim() == 5 && c10::get_if<enumtype::kBilinear>(&options.mode())) {
+  } else if (input.dim() == 5 && c10::get_if<enumtype::kBilinear>(&mode)) {
     TORCH_CHECK(false, "Got 5D input, but bilinear mode needs 4D input");
-  } else if (input.dim() == 5 && c10::get_if<enumtype::kTrilinear>(&options.mode())) {
-    return torch::upsample_trilinear3d(input, _output_size(3), *options.align_corners());
-  } else if (input.dim() == 4 && c10::get_if<enumtype::kBicubic>(&options.mode())) {
-    return torch::upsample_bicubic2d(input, _output_size(2), *options.align_corners());
+  } else if (input.dim() == 5 && c10::get_if<enumtype::kTrilinear>(&mode)) {
+    return torch::upsample_trilinear3d(input, _output_size(3), *align_corners);
+  } else if (input.dim() == 4 && c10::get_if<enumtype::kBicubic>(&mode)) {
+    return torch::upsample_bicubic2d(input, _output_size(2), *align_corners);
   } else {
     TORCH_CHECK(
         false,
         "Input Error: Only 3D, 4D and 5D input Tensors supported "
         "(got ", input.dim(), "D) for the modes: nearest | linear | bilinear | bicubic | trilinear "
-        "(got ", enumtype::get_enum_name(options.mode()), ")");
+        "(got ", enumtype::get_enum_name(mode), ")");
   }
 }
+} // namespace detail
+
+inline Tensor interpolate(const Tensor& input, InterpolateOptions options = {}) {
```

Comment:
In some of the functionals we want to be able to modify the fields in the options (such as `out` in `NormalizeFuncOptions` added in https://github.com/pytorch/pytorch/pull/29404), and Ed also suggested that we should just pass options by value because it's a small struct (https://github.com/pytorch/pytorch/pull/27942#discussion_r337604093).